src/common/snippets/src/op/subgraph.cpp

```diff
@@ -201,24 +195,54 @@ void snippets::op::Subgraph::convert_to_snippet_dialect() {
     manager.run_passes(m_body);
 }
 
-snippets::Schedule snippets::op::Subgraph::generate(const BlockedShapeVector& output_shapes, const BlockedShapeVector& input_shapes,
-                                                    ngraph::pass::Manager opt) {
+snippets::Schedule snippets::op::Subgraph::generate(const BlockedShapeVector& output_shapes,
+                                                    const BlockedShapeVector& input_shapes,
+                                                    const void* compile_params) {
+    return generate(output_shapes, input_shapes, ngraph::pass::Manager(), compile_params);
+}
+
+snippets::Schedule snippets::op::Subgraph::generate(const BlockedShapeVector& output_shapes,
+                                                    const BlockedShapeVector& input_shapes,
+                                                    ngraph::pass::Manager opt,
+                                                    const void* compile_params) {
     INTERNAL_OP_SCOPE(Subgraph);
+    OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::op::generate")
     NGRAPH_CHECK(m_generator != nullptr, "generate is called while generator is not set");
 
     canonicalize(output_shapes, input_shapes);
+
+    // Todo: ngraph::pass::Manager introduces appreciable overheads, especially while used on small graphs.
+    // So don't wrap this transformation as a MatcherPass, but rewrite convert_to_snippet_dialect() as a
+    // for loop to improve first-inference time.
+    // repalace power with power static
+
+    for (auto op : m_body->get_ordered_ops()) {
```

Comment:
So this cannot be included into convert_to_snippet_dialect just because of load time constraints?

Reply:
Yes, exactly. I don't really think that one matcher would make a difference, just thought its better to left it as-is then to rewrite it as a matcher + change it back later.

Reply:
Agree. Lets create follow up ticket.